fluency-fluentd/src/main/java/org/komamitsu/fluency/fluentd/FluencyBuilderForFluentd.java

```diff
@@ -47,7 +44,14 @@ public class FluencyBuilderForFluentd
     private SSLSocketFactory sslSocketFactory;
     protected Integer connectionTimeoutMilli;
     protected Integer readTimeoutMilli;
-    protected FluentdRecordFormatter recordFormatter = new FluentdRecordFormatter();
+    protected RecordFormatter recordFormatter;
+
+    public FluencyBuilderForFluentd() {
+        this(new FluentdRecordFormatter());
+    }
+    public FluencyBuilderForFluentd (RecordFormatter recordFormatter) {
```

Comment:
```suggestion
    public FluencyBuilderForFluentd(RecordFormatter recordFormatter) {
```

Reply:
BTW, why is this constructor needed?

Reply:
FluentdRecordFormatter depends on jackson. Through this constructor one can pass a RecordFormatter implementation that avoids this.

Reply:
The class has `setRecordFormatter()` already. It doesn't work? I don't want to make an exception for `RecordFormatter`.

Reply:
Unfortunately it doesn't work, because the `FluentdRecordFormatter` is instantiated at the time of instantiation of `FluencyBuilderForFluentd` which will cause a ClassNotFound exception for Jackson classes, if they're not found in the classpath.

Reply:
I see. How about this?

```java
    protected RecordFormatter recordFormatter;

    ...

    private RecordFormatter prepareRecordFormatter() {
        if (recordFormatter != null) {
            return recordFormatter;
        }
        else {
            return new FluentdRecordFormatter();
        }
    }

    public Fluency build(String host, int port)
    {
        return buildFromIngester(
                prepareRecordFormatter(),
                buildIngester(createBaseSender(host, port)));
    }

    public Fluency build(int port)
    {
        return buildFromIngester(
                prepareRecordFormatter(),
                buildIngester(createBaseSender(null, port)));
    }

   (Replace the following `build()` as well)
    ... 
```

Reply:
This works as well.  Personally I prefer the impl. via constructor but it's your call :)

Reply:
Thanks for the confirmation. Let's go with the `prepareRecordFormatter()`-ish way.